packages/fx-core/src/question/other.ts

```diff
@@ -994,3 +994,58 @@ export function apiSpecApiKeyQuestion(): IQTreeNode {
     ],
   };
 }
+
+export function OauthClientSecretConfirmQestion(): ConfirmQuestion {
+  return {
+    name: QuestionNames.OauthClientSecretConfirm,
+    title: getLocalizedString("core.createProjectQuestion.OauthClientSecretConfirm"),
+    type: "confirm",
+    default: true,
+  };
+}
+
+export function oauthClientSecretQuestion(): IQTreeNode {
+  return {
+    data: {
+      type: "text",
+      name: QuestionNames.OauthClientSecret,
+      cliShortName: "c",
+      title: getLocalizedString("core.createProjectQuestion.OauthClientSecret"),
+      cliDescription: "Oauth client secret for OpenAPI spec.",
+      forgetLastValue: true,
+      validation: {
+        validFunc: (input: string): string | undefined => {
+          const pattern = /^(\w){10,128}/g;
+          const match = pattern.test(input);
+
+          const result = match
+            ? undefined
+            : getLocalizedString("core.createProjectQuestion.invalidApiKey.message");
```

Comment:
rename invalidApiKey.message to invalidSecret?

Reply:
Kept here since the message is the same. Will update if we want to have difference message.